packages/state-manager/src/sagas/messages/verifyMessage/verifyMessages.saga.ts

```diff
@@ -1,14 +1,49 @@
 import { type PayloadAction } from '@reduxjs/toolkit'
-import { put } from 'typed-redux-saga'
+import { select, call, put, delay } from 'typed-redux-saga'
 import { messagesActions } from '../messages.slice'
-import { type MessageVerificationStatus } from '@quiet/types'
+import { MessageType, type MessageVerificationStatus } from '@quiet/types'
+import { publicChannelsSelectors } from '../../publicChannels/publicChannels.selectors'
+
+import { usersSelectors } from '../../users/users.selectors'
+import { verifyUserInfoMessage } from '@quiet/common'
 
 export function* verifyMessagesSaga(
   action: PayloadAction<ReturnType<typeof messagesActions.incomingMessages>>['payload']
 ): Generator {
   const messages = action.payload.messages
 
   for (const message of messages) {
+    let ownerData = yield* select(usersSelectors.ownerData)
+
+    while (true) {
+      ownerData = yield* select(usersSelectors.ownerData)
+      if (ownerData?.pubKey) {
+        break
+      }
+      yield* delay(500)
+    }
+
+    if (message.type === MessageType.Info && message.pubKey !== ownerData.pubKey) {
+      let user = yield* select(usersSelectors.getUserByPubKey(message.pubKey))
+
+      while (true) {
+        user = yield* select(usersSelectors.getUserByPubKey(message.pubKey))
+        if (user) {
+          break
+        }
+        yield* delay(500)
+      }
+      const channel = yield* select(publicChannelsSelectors.getChannelById(message.channelId))
+      if (!channel) return
+
+      const expectedMessage = yield* call(verifyUserInfoMessage, user.username, channel)
+
+      if (message.message !== expectedMessage) {
+        console.error(`${user.username} tried to send a malicious info message`)
+        return
```

Comment:
I'm not sure if you should use `return` or rather change the `isVerified` prop of the returned status object. Lacking verification status for a message may have consequences in the other parts of state-manager.